src/components/application_manager/test/commands/command_request_impl_test.cc

```diff
@@ -0,0 +1,485 @@
+#include "application_manager/smart_object_keys.h"
+#include "application_manager/application_manager.h"
+#include "application_manager/mock_application.h"
+#include "application_manager/event_engine/event.h"
+#include "application_manager/message_helper.h"
+#include "interfaces/MOBILE_API.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+
+namespace strings = ::application_manager::strings;
+namespace hmi_response = ::application_manager::hmi_response;
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::SaveArg;
+using ::testing::DoAll;
+
+using ::utils::SharedPtr;
+using ::application_manager::commands::MessageSharedPtr;
+using ::application_manager::CommandParametersPermissions;
+using ::application_manager::event_engine::EventObserver;
+using ::application_manager::commands::CommandImpl;
+using ::application_manager::commands::CommandRequestImpl;
+using ::application_manager::ApplicationManager;
+using ::application_manager::ApplicationSet;
+using ::application_manager::RPCParams;
+
+typedef ::application_manager::commands::CommandRequestImpl::RequestState
+    RequestState;
+
+class CommandRequestImplTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  class UnwrappedCommandRequestImpl : public CommandRequestImpl {
+   public:
+    using CommandRequestImpl::CheckAllowedParameters;
+    using CommandRequestImpl::RemoveDisallowedParameters;
+    using CommandRequestImpl::AddDisallowedParameters;
+    using CommandRequestImpl::HasDisallowedParams;
+
+    UnwrappedCommandRequestImpl(const MessageSharedPtr& message,
+                                ApplicationManager& application_manager)
+        : CommandRequestImpl(message, application_manager) {}
+
+    const RequestState current_state() const {
+      return current_state_;
+    }
+    void set_current_state(const RequestState state) {
+      current_state_ = state;
+    }
+
+    CommandParametersPermissions& parameters_permissions() {
+      return parameters_permissions_;
+    }
+
+    CommandParametersPermissions& removed_parameters_permissions() {
+      return removed_parameters_permissions_;
+    }
+  };
+
+  const uint32_t kConnectionKey_ = 5u;
+  const uint32_t kCorrelationId_ = 3u;
+  const hmi_apis::FunctionID::eType kInvalidFunctionId_ =
+      hmi_apis::FunctionID::INVALID_ENUM;
+  const std::string kPolicyAppId_ = "Test";
+  const mobile_apis::Result::eType kMobResultSuccess_ =
+      mobile_apis::Result::SUCCESS;
+
+  sync_primitives::Lock app_set_lock_;
+};
+
+typedef CommandRequestImplTest::UnwrappedCommandRequestImpl UCommandRequestImpl;
+typedef SharedPtr<UCommandRequestImpl> CommandPtr;
+
+TEST_F(CommandRequestImplTest, OnTimeOut_StateCompleted_UNSUCCESS) {
+  CommandPtr command = CreateCommand<UCommandRequestImpl>();
+
+  // Should be called twice:
+  // First -- on `onTimeOut` method call
+  // Second -- on destruction;
+  EXPECT_CALL(ev_disp_, remove_observer(_)).Times(2);
```

Comment:
@LevchenkoS `ev_disp_` - is event (evaluation, evacuation) display? Give more clear name to variable